src/services/submissionService.ts

```diff
@@ -0,0 +1,85 @@
+import { Submission } from '../typeDefs/dbSchema'
+import { DocumentData, WhereFilterOp, getFirestore } from 'firebase-admin/firestore'
+
+export const getSubmissionById = async (id: string) : Promise<Submission | null> => {
+    try {
+        const db = getFirestore()
+        const submissionRef = db.collection('submissions')
+        const snapshot = await submissionRef.doc(id).get()
+
+        if (!snapshot.exists) {
+            return null
+        }
+
+        const convertedEntity = mapDbEntityTogqlEntity(snapshot.data() as DocumentData)
+
+        return convertedEntity
+    } catch (error) {
+        throw new Error(`Failed to retrieve submission: ${error}.`)
+    }
+}
+
+function convertToDbSubmission(submission: import('../typeDefs/gqlTypes').Submission): 
+    import('../typeDefs/dbSchema').Submission {
+    return {
+        ...submission,
+        spokenLanguages: submission.spokenLanguages
+            .filter(lang => lang !== null) as import('../typeDefs/dbSchema').SpokenLanguage[]
+    }
+}
+
+export const addSubmission = async (submission: import('../typeDefs/gqlTypes').Submission) : Promise<Submission> => {
+    try {
+        const dbSubmission = convertToDbSubmission(submission)
+
+        const db = getFirestore()
+        const submissionRef = db.collection('submissions')
+
+        const docRef = await submissionRef.add(dbSubmission)
+
+        const newSubmission: Submission = {
+            ...dbSubmission,
+            id: docRef.id
+        }
+
+        return newSubmission
+    } catch (error) {
+        throw new Error(`Failed to add submission: ${error}.`)
+    }
+}
+
+export const updateSubmission = async (id: string, updatedFields: Partial<Submission>) : Promise<string> => {
+    try {
+        const db = getFirestore()
+        const submissionRef = db.collection('submissions').doc(id)
+
+        const snapshot = await submissionRef.get()
+
+        const submissionToUpdate = mapDbEntityTogqlEntity(snapshot.data() as DocumentData)
+
+        const updatedSubmission: Submission = {
+            ...submissionToUpdate,
+            ...updatedFields
+        }
+
+        await submissionRef.update(updatedSubmission)
+
+        return 'Submission updated successfully!'
+    } catch (error) {
+        throw new Error(`Failed to update submission: ${error}.`)
```

Comment:
you'll want to wrap the call to this method (in the resolver) with a try/catch so it doesn't shut down the API when there's an error. 
This is a good practice in general and we'll probably want all of our resolvers to follow this pattern. (the alternative design instead of throwing an Error would be that every method returns a "result" style object with a  pass/fail boolean to continue.) they both have tradeoffs. Let's start with try/catch for now since it's easier.